csrc/device_lower/pass/index.cpp

```diff
@@ -503,7 +503,7 @@ Val* getGridSyncBufferSize(
     const std::vector<ForLoop*>& for_loops,
     bool is_persistent) {
   // See the comment above for getGridCommWorkBufferSize.
-  Val* buffer_size = GpuLower::current()->kernel()->oneVal();
+  Val* buffer_size = GpuLower::current()->kernel()->oneVal(DataType::Index);
```

Comment:
This attempts to avoid introducing an `Int` when we multiply by 1 in the first step. I believe it's possible `pt_dim` below might still be `Int` sometimes, which is why I still have the maybe cast in `device_lower/utils.cpp` also.